plugin/evm/validators_normalizer_test.go

```diff
@@ -0,0 +1,66 @@
+package evm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/coreth/tests/mocks"
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/snow"
+	"github.com/flare-foundation/flare/utils/logging"
+)
+
+func TestValidatorsNormalizer_ByEpoch(t *testing.T) {
+	normalizer, copy := validatorNormalizerWithData(9, 2, true)
+
+	t.Run("nominal case", func(t *testing.T) {
+		for _, i := range []uint64{0, 1} {
+			result, err := normalizer.ByEpoch(i)
+			require.NoError(t, err)
+
+			weighted := normalizer.calcWeightRatio(copy[i])
+			assert.Equal(t, len(weighted), len(result))
+			assert.InDeltaMapValues(t, weighted, result, 0, 0)
+		}
+	})
+
+	t.Run("handles missing key", func(t *testing.T) {
+		_, err := normalizer.ByEpoch(uint64(7))
+		require.NoError(t, err)
+	})
+}
+
+func TestCalcWeightRatio(t *testing.T) {
+	normalizer, _ := validatorNormalizerWithData(9, 2, false)
+
+	validators := mocks.TestValidatorsWeightsData()
+	expected := map[ids.ShortID]uint64{
+		mocks.TestIdsList[2]: 515007957320493220,
+		mocks.TestIdsList[1]: 6465383072520688749,
+		mocks.TestIdsList[0]: 6185718675989118760,
+		mocks.TestIdsList[3]: 9599604527249266409,
+		mocks.TestIdsList[4]: 3417212044008417831,
+	}
+	weighted := normalizer.calcWeightRatio(validators)
+	assert.InDeltaMapValues(t, weighted, expected, 0, 0)
+}
```

Comment:
I guess we should also check that we do not modify the original `validators` map in here 😄